### PR TITLE
Clean up UTXOs 

### PR DESCRIPTION
**Pull request by coffeeavax**
## Why this should be merged

- Deprecates statusDB. For accepted tx, remove all consumed UTXOs in utxoDB. For other tx, remove them from txDB.
- Remove all functions writing status and using txStatus (including uniqueTxs).

### CODE REVIEW

File: vms/avm/states/state.go [hunk 1]
```diff
@@ -576,3 +578,50 @@ func (s *state) writeStatuses() error {
...
+		if choices.Status(valStatus) == choices.Rejected {
+			txID, err := ids.ToID(key)
+			if err != nil {
+				return []ids.ID{}, err
+			}
+			txIDs = append(txIDs, txID)
+			// remove txID from dbStatus
+			if err := database.ClearPrefix(s.statusDB, s.statusDB, key); err != nil {
+				return []ids.ID{}, err
+			}
```
Comment (StephenButtolph) [inline vms/avm/states/state.go:601]: Why are we using `database.ClearPrefix`? This is just removing a single key. `s.statusDB.Delete(key)`.

File: vms/avm/states/state.go [hunk 2]
```diff
@@ -576,3 +578,50 @@ func (s *state) writeStatuses() error {
...
+	if err != nil {
+		return err
+	}
+	for _, txID := range txIDs {
+		tx, err := s.GetTx(txID)
+		if err != nil {
+			return err
+		}
+		// find UTXOID produced by the tx
+		utxos := tx.UTXOs()
```
Comment (StephenButtolph) [inline vms/avm/states/state.go:618]: We should not be touching produced UTXOs.

**Review by StephenButtolph — CHANGES_REQUESTED** [review]

File: vms/avm/states/state.go (hunk 2 above)
Comment (coffeeavax) [reply vms/avm/states/state.go]: updated

File: vms/avm/states/state.go [hunk 3]
```diff
@@ -576,3 +578,50 @@ func (s *state) writeStatuses() error {
 	}
 	return nil
 }
+
+func (s *state) rejectedTxs() ([]ids.ID, error) {
+	var txIDs []ids.ID
+	iter := s.statusDB.NewIterator()
```
[the remark this answers is not in the document]
Comment (coffeeavax) [reply vms/avm/states/state.go:585]: updated

File: vms/avm/states/state.go (hunk 1 above)
Comment (coffeeavax) [reply vms/avm/states/state.go]: updated.

**Pull request closed without merging**